src/FakeItEasy/Core/FakeManagerProvider.cs

```diff
@@ -81,13 +81,19 @@ public void EnsureInitialized(object proxy)
                 {
                     this.initializedFakeManager = this.fakeManagerFactory(this.typeOfFake, proxy);
 
-                    this.fakeManagerAccessor.TagProxy(proxy, this.initializedFakeManager);
+                    this.fakeManagerAccessor.SetFakeManager(proxy, this.initializedFakeManager);
 
                     this.ApplyInitialConfiguration(proxy);
                 }
             }
         }
 
+        public void EnsureManagerIsRegistered()
+        {
+            var accessor = ServiceLocator.Current.Resolve<IFakeManagerAccessor>();

```

Comment:
I think I tried, but it didn't work. But I changed other things since then. I'll see what I can do. It would definitely be cleaner.